### PR TITLE
Fix, Docs, Test: Fibonacci_fast (Renamed to Fibonacci_iterative)

### PR DESCRIPTION
**Pull request by Renjian-buchai**
#### Description of Change

The original algorithm does not calculate the nth fibonacci number, but rather the next fibonacci number after storing it in a static variable. Thus, I abstracted the looping part out of the code that you have to write upon application, into the function itself.

The "Test" in main does not really test anything, rather demonstrates what it does. Added actual unit tests with test cases.

The "Test" in main also does not calculate the nth fibonacci number, but rather the (n+1)th fibonacci number.

The maximum value that can be computed such that the result is less than UINT64_MAX is also incorrect because of the above error. 

The file brief did not explain what it actually does properly.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
Fixed bugs, added documentation, and unit tests for Fibonacci_fast (Renamed to Fibonacci_iterative to better reflect what it actually does). 

For #2456.

### CODE REVIEW

**Comment by Renjian-buchai** [comment]
I genuinely don't know what the original author was trying to do here, tbh.

File: math/fibonacci_iterative.cpp [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    std::cout << "All tests passed successfully.";
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    test();
```
Comment (realstealthninja) [inline math/fibonacci_iterative.cpp:126]:
```suggestion
    test(); // run self-test implementations
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Review by realstealthninja — APPROVED** [review]
LGTM

**Comment by realstealthninja** [comment]
@Panquesito7  could we get this merged?

**Review by realstealthninja — APPROVED** [review]

**Pull request closed without merging**